src/binary/timestamp.rs

```diff
@@ -87,41 +87,74 @@ where
                                     let scaled = utc.nanosecond() / 10u32.pow(9 - *precision); // 123,000,000 -> 123
                                     let exponent = (*precision as i64).neg(); // -3
                                     let fractional = Decimal::new(scaled, exponent); // 123d-3
-                                    self.encode_decimal(&fractional)?;
+                                    bytes_written += self.encode_decimal(&fractional)?;
+                                }
+                                Mantissa::Arbitrary(decimal) => {
+                                    bytes_written += self.encode_decimal(decimal)?;
                                 }
-                                Mantissa::Arbitrary(decimal) => self.encode_decimal(decimal)?,
                             };
                         }
                     }
                 }
             }
         }
 
-        Ok(())
+        Ok(bytes_written)
     }
 
-    fn encode_timestamp_value(&mut self, timestamp: &Timestamp) -> IonResult<()> {
+    fn encode_timestamp_value(&mut self, timestamp: &Timestamp) -> IonResult<usize> {
+        let mut bytes_written: usize = 0;
+
         // First encode the timestamp. We need to know the encoded length before
         // we can compute and write out the type descriptor.
         let mut encoded: ArrayVec<u8, MAX_TIMESTAMP_LENGTH> = ArrayVec::new();
-        encoded.encode_timestamp(timestamp)?;
+        bytes_written += encoded.encode_timestamp(timestamp)?;
 
         // Write the type descriptor and length.
         let type_descriptor: u8;
         if encoded.len() <= MAX_INLINE_LENGTH {
             type_descriptor = 0x60 | encoded.len() as u8;
             self.write(&[type_descriptor])?;
+            bytes_written += 1;
         } else {
             type_descriptor = 0x6E;
             self.write(&[type_descriptor])?;
-            VarUInt::write_u64(self, encoded.len() as u64)?;
+            bytes_written += 1;
+            bytes_written += VarUInt::write_u64(self, encoded.len() as u64)?;
         }
 
         // Now we can write out the encoded timestamp!
         self.write(&encoded[..])?;
+        bytes_written += &encoded[..].len();
 
-        Ok(())
+        Ok(bytes_written)
     }
 }
 
-// Looking for the tests? See binary/writer.rs.
+#[cfg(test)]
+mod binary_timestamp_tests {
+    use super::*;
+    use chrono::DateTime;
+    use rstest::*;
+
+    // These tests show how varying levels of precision affects number of bytes
+    // written (for binary encoding of timestamps).
+    #[rstest]
+    #[case::y2k_utc("2000-01-01T00:00:00+00:00", 19)]
```

Comment:
Easy answer - that's because I had a bug where I was doublecounting the representation.

That said, my latest commit doesn't agree with the above output.

```
2000-01-01T00:00:00+00:00 -> [69, 80, 0f, d0, 81, 81, 80, 80, 80, c9]
```

So the first type is 68 not 69 (because my output has +1 byte) and the last (extra) byte is c9. I think that's because the above example shows up with 0d-9 nanoseconds. So... looks like 2 bugs! If you don't mind, I'll open (and own) and issue for the second one (treat is as orthogonal to this PR).

Reply:
That works, thanks!